src/api/dataLoader.tsx

```diff
@@ -0,0 +1,18 @@
+import { IData } from "../interfaces/IData";
+
+class dataLoader {
+  loadData(): Promise<IData[]> {
+    return new Promise((resolve, reject) => {
+      fetch("https://rickandmortyapi.com/api/character")
+        .then((response) => response.json())
+        .then((data) => {
+          resolve(data.results);
+        })
+        .catch((error) => {
+          reject(error);
+        });
+    });
+  }
+}
```

Comment:
You can add to class above and combine them. 
And also try to use static methods